File: 83.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        dummy = ListNode(0)
        dummy.next = head
        pre = dummy
        while pre and pre.next and pre.next.next:
            if pre.next.val == pre.next.next.val:
                pre.next = pre.next.next
            else:
                pre = pre.next
        return dummy.next

File: 83.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* dummy = new ListNode(0);
        dummy->next = head;
        ListNode* pre = dummy;
        while(pre and pre->next and pre->next->next){
            if (pre->next->val == pre->next->next->val){
                pre->next = pre->next->next;
            }else{
                pre = pre->next;
            }
        }
        return dummy->next;
    }
};
